helper/dbtxn/dbtxn.go

```diff
@@ -0,0 +1,109 @@
+package dbtxn
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+type Config struct {
+	Name       string
+	Username   string
+	Password   string
+	Expiration string
+}
+
+// ExecuteDBQuery handles executing one single statement, while properly releasing its resources.
+// - ctx: 	Optional, may be nil
+// - db: 	Required
+// - config: 	Optional, may be nil
+// - query: 	Required
+func ExecuteDBQuery(ctx context.Context, db *sql.DB, config *Config, query string) error {
+
+	parsedQuery := parseQuery(config, query)
+
+	var stmt *sql.Stmt
+	var err error
+	if ctx != nil {
+		stmt, err = db.PrepareContext(ctx, parsedQuery)
+	} else {
+		stmt, err = db.Prepare(parsedQuery)
+	}
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	return execute(ctx, stmt)
+}
+
+// ExecuteTxQuery handles executing one single statement, while properly releasing its resources.
+// - ctx: 	Optional, may be nil
```

Comment:
How come context would be nil?

Reply:
It was originally because it wasn't always used, but on your suggestion below, I flipped everything over to using context and stripped out the nil check.